### PR TITLE
Updates for discovery 3.0

### PR DESCRIPTION
**Pull request by lzap**

### CODE REVIEW

**Comment by domcleal** [comment]
This is the 2.0 doc, did you want to separate it?

**Comment by lzap** [comment]
> This is the 2.0 doc, did you want to separate it?

Yeah, there are no big changes tho. That confused me.

## 

Later,
 Lukas #lzap Zapletal

File: plugins/foreman_discovery/3.0/index.md [hunk 1]
```diff
@@ -0,0 +1,679 @@
+---
+layout: plugin
+title: Foreman Discovery 2.0 Manual
```
Comment (domcleal) [inline plugins/foreman_discovery/3.0/index.md]: 3.0, and the line below

File: plugins/foreman_discovery/3.0/index.md [hunk 2]
```diff
@@ -0,0 +1,679 @@
...
+    <td>3.0.0</td>
+    <td>1.0.2</td>
+    <td>2.1</td>
+    <td>0.0.1</td>
+  </tr>
+</table>
+
+### 1.1.1 Foreman Discovery plugin
+
+**2.0**: New features:
```
Comment (domcleal) [inline plugins/foreman_discovery/3.0/index.md:96]: 3.0 notes needed here

File: plugins/foreman_discovery/3.0/index.md [hunk 3]
```diff
@@ -0,0 +1,679 @@
...
+* new CLI hammer plugin
+* reboot option
+
+### 1.1.2 Foreman Proxy Discovery plugin
+
+**1.0**: Initial version.
+
+### 1.1.3 Foreman Discovery Image
+
+**2.0**: New features:
```
Comment (domcleal) [inline plugins/foreman_discovery/3.0/index.md]: 2.1?

Comment (domcleal) [inline plugins/foreman_discovery/3.0/index.md:117]: Add 2.1 notes?

**Comment by domcleal** [comment]
Also needs linking from plugins/index.html.

**Comment by lzap** [comment]
> Also needs linking from plugins/index.html.

All fixed.

## 

Later,
 Lukas #lzap Zapletal

File: plugins/foreman_discovery/3.0/index.md [hunk 4]
```diff
@@ -0,0 +1,689 @@
...
+    --
+    Discovered Host > Foreman Proxy > Foreman
+    --
+    Foreman > Foreman Proxy > Discovered Host
+    --
+
+Foreman Discovery does support "legacy" mode where communication is direct
+with no Foreman Proxy involved. Currently (Foreman Discovery 3.0) this is
+still the recommended way to configure it due to technical limitations of
+Foreman Proxy 1.7 (this is explained below).
```
Comment (domcleal) [inline plugins/foreman_discovery/3.0/index.md]: Out of date?

File: plugins/foreman_discovery/3.0/index.md [hunk 5]
```diff
@@ -0,0 +1,689 @@
...
+Proceed with the next step (Reboot discovered hosts) and in the next section
+(Configuration) make sure the kernel line in the PXELinux default template has
+`proxy.type` set to `foreman` and `proxy.url` points to a *Foreman* instance.
+
+    APPEND ... proxy.url=https://YOUR_FOREMAN proxy.type=foreman
+
+### 2.5.2 Upgrade with Proxy Discovery plugin
+
+This method is *not recommended* for Discovery 2.0 because due to limitation
+of Foreman Proxy 1.7 it is not possible to configure this via HTTPS protocol.
```
Comment (domcleal) [inline plugins/foreman_discovery/3.0/index.md]: outdated?

File: plugins/foreman_discovery/3.0/index.md [hunk 6]
```diff
@@ -0,0 +1,689 @@
...
+
+## 2.5 Upgrade
+
+To upgrade Foreman Discovery follow the standard procedure of upgrading all
+the Foreman packages.
+
+New component Proxy Discovery plugin was introduced for Discovery 2.0
+therefore there are multiple ways how to re-configure existing installation.
+
+### 2.5.1 Upgrade without Proxy Discovery plugin (recommended)
```
Comment (domcleal) [inline plugins/foreman_discovery/3.0/index.md]: no longer recommended?

Comment (lzap) [inline plugins/foreman_discovery/3.0/index.md]: > no longer recommended?

Yeah, I wanted to wait one additional release with that, but I've
doublechecked and there is no need to wait. Changed everything - we now
default to proxy configuration.

## 

Later,
 Lukas #lzap Zapletal

**Comment by domcleal** [comment]
Thanks @lzap!

**Pull request merged**